sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnAlertSystemWindowsTest.java

```diff
@@ -295,5 +300,70 @@ public void returningToDialogFromBackgroundWhenInstallerIsNull() {
         AppCenterLog.debug(eq(LOG_TAG), eq("Installing couldn't start due to the release installer wasn't initialized."));
     }
 
+    @Test
+    public void showSettingsDialogWhenActivityIsNull() throws Exception {
+
+        /* Mock permission state for Android Q. */
+        when(InstallerUtils.isSystemAlertWindowsEnabled(any(Context.class))).thenReturn(false);
+        when(Settings.canDrawOverlays(any(Context.class))).thenReturn(false);
+
+        /* Set Activity NULL */
+        Distribute.getInstance().onActivityPaused(mActivity);
+
+        Mockito.when(DistributeUtils.getStoredDownloadState()).thenReturn(DOWNLOAD_STATE_NOTIFIED);
+
+        /* Mock notification manager to avoid NRE. */
+        NotificationManager manager = Mockito.mock(NotificationManager.class);
+        whenNew(Notification.Builder.class).withAnyArguments()
+                .thenReturn(Mockito.mock(Notification.Builder.class, RETURNS_MOCKS));
+        Mockito.when(mContext.getSystemService(NOTIFICATION_SERVICE)).thenReturn(manager);
+
+        /* Notify about complete download. */
+        mReleaseDownloaderListener.onComplete(1L);
+
+        /* Verify that dialog was not shown. */
+        verify(mAlertWindowsDialog, never()).show();
+
+        /* Verify system alert window is not trying to display if activity is null */
+        verifyStatic();
+        AppCenterLog.warn(eq(LOG_TAG), eq("The application is in background mode, the system alerts windows won't be displayed."));
+    }
+
+    @Test
+    public void needRefreshDialogWhenStartInstallation() throws Exception {
+
+        /* Mock permission state for Android Q. */
+        when(InstallerUtils.isSystemAlertWindowsEnabled(any(Context.class))).thenReturn(false);
+        when(Settings.canDrawOverlays(any(Context.class))).thenReturn(false);
+
+        Mockito.when(DistributeUtils.getStoredDownloadState()).thenReturn(DOWNLOAD_STATE_NOTIFIED);
+
+        /* Mock notification manager to avoid NRE. */
+        NotificationManager manager = Mockito.mock(NotificationManager.class);
+        whenNew(Notification.Builder.class).withAnyArguments()
+                .thenReturn(Mockito.mock(Notification.Builder.class, RETURNS_MOCKS));
+        Mockito.when(mContext.getSystemService(NOTIFICATION_SERVICE)).thenReturn(manager);
+
+        /* Notify about complete download. */
+        mReleaseDownloaderListener.onComplete(1L);
+
+        /* Start distribute with app secret NULL to make sure updateReleaseDetails is called on
+           startFromBackground.
+         */
+        Distribute.getInstance().onStarted(mContext, mChannel, null, null, true);
+        Distribute.getInstance().startFromBackground(mContext);
+
+        /* isMandatoryUpdate should be mocked to avoid completion installation workflow */
```

Comment:
`isMandatoryUpdate should be mocked to avoid completion installation workflow` -> `Release details should be mandatory to avoid completion installation workflow.`